odoo/addons/base/models/ir_model.py

```diff
@@ -879,6 +879,20 @@ def _drop_column(self):
 
         return True
 
+    def _get_inherit_to_remove(self, modules_to_remove):
+        modules_to_remove = set(modules_to_remove)
+        model_name_to_field_records = self.grouped('model')
+        inherit_fields_to_remove = self.browse()
+        for model_name, field_records in model_name_to_field_records.items():
+            if model_name not in self.env:
+                continue
+            for model_name_ in self.pool[model_name]._inherit_children:
```

Comment:
`_inherit_children` is not a transitive closure, it's only the "direct" children models.
So I guess you should recurse at some point...